test/e2e/framework/framework.go

```diff
@@ -317,17 +320,31 @@ func (f *Framework) AfterEach() {
 	}
 
 	outputTypes := strings.Split(TestContext.OutputPrintType, ",")
+	now := time.Now()
 	for _, printType := range outputTypes {
 		switch printType {
 		case "hr":
 			for i := range summaries {
-				Logf(summaries[i].PrintHumanReadable())
+				if TestContext.ReportDir == "" {
+					Logf(summaries[i].PrintHumanReadable())
+				} else {
+					filePath := path.Join(TestContext.ReportDir, summaries[i].SummaryKind()+now.Format(time.RFC3339)+".txt")
```

Comment:
Done. It may be tricky though, as you certainly don't want to append "should allow starting 30 pods per node using { ReplicationController} with 0 secrets and 0 daemons" (aka "Density") which is what you would get...